### DISCUSSION TITLE
Error when run "npm ci"

### DISCUSSION THREAD

**Discussion by jasoncks** [mozilla/hubs#3291] [Hubs Cloud Questions] [answered]
In the past, it is possible for my PC to run "npm ci" without any error.
Recently, I did not install any new software or changing the setting of the repos/github, even I try to run "npm ci" in "master"/ "hubs-cloud" batch using the source in "mozilla/hubs" repos, some errors are shown in red.
![hubs_npm_ci](https://user-images.githubusercontent.com/25451208/98466899-c59d2580-2215-11eb-85cf-43737b8892e4.PNG)
But it seems that it does not actually an error since Hubs still run/build normally.
How can I solve the error?

**Comment by robinkwilson** [comment]
Are you still seeing this error + is this hindering your progress? What OS system are you using Windows, OSX, or Linux?

**Reply by jasoncks** [reply]
The error is still occurring. 
I am using Windows10 for my development.

This error is no t hindering my progress.
I can still build, run, deploy Hubs.

But I just want to confirm if these errors are really harmless?

**Reply by robinkwilson** [reply]
What `node` version are you running? 

Seems to be an error in a dependency package, glad it isn't hindering your progress. 

(I'm on a Mac and am on node v13.13.0 and npm 6.14.5. One of our team members on Windows had to update their node version to get things to run.)

**Reply by robinkwilson** [reply]
Marking as answer to remove thread from our pipeline, feel free to "Unmark as answer" when you add more information!

**Reply by jasoncks** [reply]
Sorry for the late reply
I have updated node and npm version, but the problem still occurs
node v14.15.3
npm 6.14.8

**Comment by Roboterson** [comment]
I seem to be having the same issue as seen here, and I think it is what is not allowing me to deploy to my hubs server. It looks like the version of websocket may not be correct. What is the correct version that I should be using as my version is websocket@1.0.31.

**Reply by robinkwilson** [reply]
What `node` version are you running?
What `npm` version are your running?
Can you send me the error print out from running `npm run deploy`?

**Reply by Roboterson** [reply]
Node: [v15.0.1]([url](url))
npm: 6.14.8
Errors reported when running in console. Is this enough or should I also add the debug log?
`C:\CODING_WORK\Work\Hub_VR\hubs>npm run deploy

> hubs@0.0.1 deploy C:\CODING_WORK\Work\Hub_VR\hubs
> node -r @babel/register -r esm -r ./scripts/shim scripts/deploy.js

Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
Deploying to metahood.ca.
  \ Building Client.Error: Command failed: npm ci
npm ERR! code 1
npm ERR! git dep preparation failed
npm ERR! command C:\Program Files\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\bin\npm-cli.js install --only=dev --prod --ignore-prepublish --no-progress --no-save --cache=C:\Users\MY_USER\AppData\Roaming\npm-cache\_cacache --prefer-offline=false --prefer-online=false --offline=false --before=
npm ERR! > networked-aframe@0.6.1 prepublish
npm ERR! > npm run dist
npm ERR!
npm ERR!
npm ERR! > networked-aframe@0.6.1 dist
npm ERR! > webpack --config webpack.config.js && webpack -p --config webpack.prod.config.js
npm ERR! npm WARN invalid config before="" set in command line options
npm ERR! npm WARN invalid config Must be one of: null, valid Date string
npm ERR! 'webpack' is not recognized as an internal or external command,
npm ERR! operable program or batch file.
npm ERR! npm ERR! code 1
npm ERR! npm ERR! path C:\Users\MY_USER\AppData\Roaming\npm-cache\_cacache\tmp\git-clone-b2974f3a
npm ERR! npm ERR! command failed
npm ERR! npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c "webpack --config webpack.config.js && webpack -p --config webpack.prod.config.js"
npm ERR!
npm ERR! npm ERR! A complete log of this run can be found in:
npm ERR! npm ERR!     C:\Users\MY_USER\AppData\Roaming\npm-cache\_cacache\_logs\2021-01-08T00_26_13_765Z-debug.log
npm ERR! npm ERR! code 1
npm ERR! npm ERR! path C:\Users\MY_USER\AppData\Roaming\npm-cache\_cacache\tmp\git-clone-b2974f3a
npm ERR! npm ERR! command failed
npm ERR! npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c "npm run dist"
npm ERR!
npm ERR! npm ERR! A complete log of this run can be found in:
npm ERR! npm ERR!     C:\Users\MY_USER\AppData\Roaming\npm-cache\_cacache\_logs\2021-01-08T00_26_13_801Z-debug.log

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\MY_USER\AppData\Roaming\npm-cache\_logs\2021-01-08T00_26_58_873Z-debug.log

    at ChildProcess.exithandler (node:child_process:309:12)
    at ChildProcess.emit (node:events:327:20)
    at ChildProcess.EventEmitter.emit (node:domain:486:12)
    at maybeClose (node:internal/child_process:1048:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:288:5) {
  killed: false,
  code: 1,
  signal: null,
  cmd: 'npm ci'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! hubs@0.0.1 deploy: `node -r @babel/register -r esm -r ./scripts/shim scripts/deploy.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the hubs@0.0.1 deploy script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\MY_USER\AppData\Roaming\npm-cache\_logs\2021-01-08T00_26_59_844Z-debug.log`

**Comment by robinkwilson** [comment, marked as answer]
Can you try running that command prompt as "admin"? 
Right click the command prompt icon to open the app, then select run as admin.

Another user was having issues on PC and this was their fix. Let's see if it works for you!

**Reply by robinkwilson** [reply]
Marking as answered for now, if you're still having issues please unmark as answer with new info! Thank you!
